site/content/integrate/apps/using-third-party-api/hello-oauth2/_index.md

```diff
@@ -0,0 +1,316 @@
+---
+title: "Oauth2"
+heading: "Hello, Oauth2!"
+description: "In this example, the http will connect via oauth2"
+weight: 90
+---
+
+This is an example of an HTTP app ([source](https://github.com/mattermost/mattermost-plugin-apps/tree/master/examples/go/hello-oauth2)), written in Go and runnable on http://localhost:8080.
+
+- It contains a `manifest.json`, declares itself an HTTP application, requests permissions, and binds itself to locations in the Mattermost user interface.
+- In its `bindings` function it declares three commands: `configure`, `connect`, and `send`.
+- Its `send` function mentions the user by their Google name, and lists their Google Calendars.
+
+To install "Hello, OAuth2" on a locally-running instance of Mattermost follow these steps (go 1.16 is required):
+
+```sh
+git clone https://github.com/mattermost/mattermost-plugin-apps.git
+cd mattermost-plugin-apps/examples/go/hello-oauth2
+go run . 
+```
+
+In the Mattermost Desktop client run:
+
+```
+/apps install http http://localhost:8080/manifest.json
+```
+
+You need to configure your [Google API Credentials](https://console.cloud.google.com/apis/credentials) for the app. Use `$MATTERMOST_SITE_URL$/com.mattermost.apps/apps/hello-oauth2/oauth2/remote/complete` for the `Authorized redirect URIs` field. After configuring the credentials, in the Mattermost Desktop client run:
+
+```
+/hello-oauth2 configure --client-id $CLIENT_ID --client-secret $CLIENT_SECRET
+```
+
+Now, you can connect your account to Google with `/hello-oauth2 connect` command, and then try `/hello-oauth2 send`.
+
+### Manifest
+
+Hello OAuth2! is an HTTP app, it requests the *permissions* to act as a System Admin to change the app's OAuth2 config, as a user to connect and send. It binds itself to `/` commands.
+
+```json
+{
+    "app_id": "hello-oauth2",
+    "version":"demo",
+    "display_name": "Hello, OAuth2!",
+    "app_type": "http",
+    "root_url": "http://localhost:8080",
+    "homepage_url": "https://github.com/mattermost/mattermost-plugin-apps/examples/go/hello-oauth2",
```

Comment:
That's a 404

Reply:
[`95f1d87` (#828)](https://github.com/mattermost/mattermost-developer-documentation/pull/828/commits/95f1d8717f4cc9db0037f0d294f68d32f801c01b)